src/v/cluster/controller_stm.cc

```diff
@@ -239,4 +239,16 @@ ss::future<std::error_code> controller_stm::replicate_and_wait(
       });
 }
 
+ss::future<result<model::offset>> controller_stm::insert_linearizable_barrier(
```

Comment:
i'm guessing the commits should be reordered, otherwise the first won't compile?

Reply:
why? the first one is just to reduce code duplication

Reply:
i think you read/answered this question after your force push